partial implementation include all symmetries

MaMiFreak opened the pull request.

MaMiFreak committed:
partial implementation include all symmetries

MaMiFreak committed:
Update supervised_policy_trainer.py

wrongu commented on AlphaGo/training/supervised_policy_trainer.py:
```diff
@@ -85,6 +85,57 @@ def on_epoch_end(self, epoch, logs={}):
 }
 
 
+def load_indices_from_file(symmetries, shuffle_file):
```
What does this function do? (`symmetries` not used?)

wrongu commented on AlphaGo/training/supervised_policy_trainer.py:
```diff
@@ -85,6 +85,57 @@ def on_epoch_end(self, epoch, logs={}):
+    return indices
+
+
+def set_indices(args, n_total_data_size, symmetries, shuffle_file_train, shuffle_file_val):
```
Replace `args` with each of `max_validation` and `train_val_test`

wrongu replied:
Also, would it make sense for this function to always generate an array with all 8 symmetries? What happens if I (for some reason) start training with `--symmetries=none` but later resume training with `--symmetries=all`?

wrongu replied:
Last comment here: I'd rename this function to `create_and_save_shuffle_indices`

wrongu commented on AlphaGo/training/supervised_policy_trainer.py:
```diff
@@ -85,6 +85,57 @@ def on_epoch_end(self, epoch, logs={}):
+    # shuffle array
+    np.random.shuffle(shuffle_indices)
+
+    print(len(shuffle_indices))
```
Either remove these `print`s or wrap them in a `verbose` check and describe them, e.g. `print("created %d total shuffle indices for augmented data from %d positions and %d symmetries" % (len(shuffle_indices), n_total_data_size, len(symmetries))`

wrongu commented on AlphaGo/training/supervised_policy_trainer.py:
```diff
@@ -85,6 +85,57 @@ def on_epoch_end(self, epoch, logs={}):
+
+def set_indices(args, n_total_data_size, symmetries, shuffle_file_train, shuffle_file_val):
+
+    # create array with all position, symmetry combinations
```
Use a more descriptive comment here. For example:

`# Create an array with a unique row for each combination of a training example and a symmetry. shuffle_indices[i][0] is an index into training examples, and shuffle_indices[i][1] is the index (from 0 to 7) of the symmetry transformation to apply.`

wrongu commented on AlphaGo/training/supervised_policy_trainer.py:
```diff
@@ -85,6 +85,57 @@ def on_epoch_end(self, epoch, logs={}):
+
+    # create array with all position, symmetry combinations
+    shuffle_indices = np.empty(shape=[n_total_data_size * len(symmetries), 2], dtype=int)
+    for x in range(n_total_data_size):
```
instead of `x` and `y`, use a descriptive name like `dataset_idx` and `symmetry_idx`

wrongu commented on AlphaGo/training/supervised_policy_trainer.py:
```diff
@@ -85,6 +85,57 @@ def on_epoch_end(self, epoch, logs={}):
+        n_val_data = args.max_validation
+
+    # train set size
+    n_train_data = len(shuffle_indices) - n_val_data
```
Leave room for the test set

wrongu commented on AlphaGo/training/supervised_policy_trainer.py:
```diff
@@ -85,6 +85,57 @@ def on_epoch_end(self, epoch, logs={}):
+            shuffle_indices[x * len(symmetries) + y][0] = x
+            shuffle_indices[x * len(symmetries) + y][1] = y
+
+    # shuffle array
```
I'd clarify in the comment briefly that this shuffles rows without affecting `x,y` pairs

wrongu commented on AlphaGo/training/supervised_policy_trainer.py:
```diff
@@ -85,6 +85,57 @@ def on_epoch_end(self, epoch, logs={}):
+
+    # training set
+    train_indices = shuffle_indices[0:n_train_data]
+    # train_indices = shuffle_indices[shuffle_indices[:,1] <= n_train_data, :]
```
Please remove commented code

wrongu commented on AlphaGo/training/supervised_policy_trainer.py:
```diff
@@ -33,9 +33,9 @@ def shuffled_hdf5_batch_generator(state_dataset, action_dataset,
             # get state from dataset and transform it.
```
Need to update transform from `np.random.choice` to `transforms[data_idx[1]]`. __But__ then you should also create a canonical ordering of the transformations; don't assume that `BOARD_TRANSFORMATIONS.keys()` will always be in the same order. We might need a new constant

```python
TRANSFORMATION_INDICES = {
'noop': 0,
'rot90': 1,
'rot180': 2,
...
}
```

which would then need to be used both in the function that loads indices from files and in the function that creates and saves them.

wrongu reviewed (changes requested):
I like the idea! Mostly some nit-picking things about comments and variable names. Looks like two of the functions are unfinished too, but I'm sure you're working on that

MaMiFreak committed:
symmetries implemented, additional comments

feels like some checks have to be added to verify the same .hdf5 feature file is used and the same amount of symmetries

MaMiFreak committed:
update supervised training test

MaMiFreak committed:
corrected training set size

MaMiFreak committed:
trailing whitespaces

MaMiFreak committed:
solved flake8 errors

MaMiFreak committed:
flake8 fix

MaMiFreak committed:
flake8 fix

MaMiFreak committed:
flak8 correction

MaMiFreak committed:
var update

MaMiFreak committed:
check data file and state amount

MaMiFreak committed:
added symmertries count test

MaMiFreak committed:
updated comments

MaMiFreak committed:
added resume learning test

MaMiFreak committed:
added white spaces

MaMiFreak committed:
.append() moved to new line

MaMiFreak committed:
removed whitespaces

MaMiFreak committed:
def structure change

MaMiFreak committed:
removed whitespaces

MaMiFreak committed:
first check with travis

MaMiFreak committed:
second travis check

MaMiFreak committed:
Update test_supervised_policy_trainer.py

MaMiFreak committed:
Update supervised_policy_trainer.py

MaMiFreak committed:
Update supervised_policy_trainer.py

MaMiFreak committed:
Update supervised_policy_trainer.py

MaMiFreak committed:
Update supervised_policy_trainer.py

MaMiFreak committed:
Update supervised_policy_trainer.py

MaMiFreak committed:
Update supervised_policy_trainer.py

MaMiFreak committed:
Update supervised_policy_trainer.py

MaMiFreak committed:
Update supervised_policy_trainer.py

MaMiFreak committed:
Update supervised_policy_trainer.py

MaMiFreak committed:
Update supervised_policy_trainer.py

MaMiFreak committed:
Update supervised_policy_trainer.py

MaMiFreak committed:
keras update 1.2.0 and numpy 1.11.3

MaMiFreak committed:
keras update 1.2.0 and numpy 1.11.3

MaMiFreak committed:
Update .travis.yml

MaMiFreak committed:
Revert "Update .travis.yml"

This reverts commit 2a3528a488e93eb178c9e1dbacf1bd14b7434f0c.

MaMiFreak committed:
Revert "keras update 1.2.0 and numpy 1.11.3"

This reverts commit d5854001f09b81efc1d7c0084ca5b0f1fac58733.

MaMiFreak committed:
Revert "keras update 1.2.0 and numpy 1.11.3"

This reverts commit b90aae79a4e99dfc644a3fcc831fc374f7e29dbb.
